utils/regression-analysis/src/main.rs

```diff
@@ -0,0 +1,221 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2022 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    collections::BTreeMap,
+    fs,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+use clap::{Parser, Subcommand};
+
+use quick_xml::de::from_str;
+
+mod junit_tree;
+mod output;
+
+use common::TestSuites;
+use tabled::{Style, Table};
+
+const PALLET_NAMES: [&str; 7] = [
+    "pallet-gas",
+    "pallet-gear",
+    "pallet-gear-debug",
+    "pallet-gear-messenger",
+    "pallet-gear-program",
+    "pallet-gear-payment",
+    "pallet-usage",
+];
+
+const PREALLOCATE: usize = 1_000;
+
+const TEST_SUITES_TEXT: &str = "Test suites";
+
+#[derive(Parser)]
+struct Cli {
+    #[clap(subcommand)]
+    command: Commands,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    CollectData {
+        #[clap(long)]
+        data_folder_path: PathBuf,
+        #[clap(long)]
+        output_path: PathBuf,
+        #[clap(long)]
+        disable_filter: bool,
+    },
+    Compare {
+        #[clap(long)]
+        data_path: PathBuf,
+        #[clap(long)]
+        current_junit_path: PathBuf,
+        #[clap(long)]
+        disable_filter: bool,
+    },
+}
+
+fn build_tree<P: AsRef<Path>>(
+    disable_filter: bool,
+    path: P,
+) -> BTreeMap<String, BTreeMap<String, f64>> {
+    let filter = |pallet_name: &str| {
+        if disable_filter {
+            return true;
+        }
+
+        PALLET_NAMES.iter().any(|&name| name == pallet_name)
+    };
+
+    let junit_xml = std::fs::read_to_string(path).unwrap();
+    let test_suites: TestSuites = from_str(&junit_xml).unwrap();
+    let total_time: BTreeMap<_, _> = [(
```

Comment:
```suggestion
    let total_time = [(
```

Isn't it may be skipped?

Reply:
yep